src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/listeners/AssessmentListener.kt

```diff
@@ -13,20 +13,34 @@ class AssessmentListener {
   @PrePersist
   fun prePersist(assessment: ApprovedPremisesAssessmentEntity) {
     if (assessment.allocatedToUser == null) {
-      (assessment.application as ApprovedPremisesApplicationEntity).status = ApprovedPremisesApplicationStatus.UNALLOCATED_ASSESSMENT
+      (assessment.application as ApprovedPremisesApplicationEntity).status =
+        ApprovedPremisesApplicationStatus.UNALLOCATED_ASSESSMENT
     } else {
-      (assessment.application as ApprovedPremisesApplicationEntity).status = ApprovedPremisesApplicationStatus.AWAITING_ASSESSMENT
+      (assessment.application as ApprovedPremisesApplicationEntity).status =
+        ApprovedPremisesApplicationStatus.AWAITING_ASSESSMENT
     }
   }
 
   @PreUpdate
   fun preUpdate(assessment: ApprovedPremisesAssessmentEntity) {
     if (assessment.decision == null && assessment.data != null) {
-      (assessment.application as ApprovedPremisesApplicationEntity).status = ApprovedPremisesApplicationStatus.ASSESSMENT_IN_PROGRESS
+      (assessment.application as ApprovedPremisesApplicationEntity).status =
+        ApprovedPremisesApplicationStatus.ASSESSMENT_IN_PROGRESS
     } else if (assessment.decision == AssessmentDecision.ACCEPTED) {
-      (assessment.application as ApprovedPremisesApplicationEntity).status = ApprovedPremisesApplicationStatus.AWAITING_PLACEMENT
+      val application = (assessment.application as ApprovedPremisesApplicationEntity)
+      application.status = applicationGetStatusFromArrivalDate(application)
     } else if (assessment.decision == AssessmentDecision.REJECTED) {
       (assessment.application as ApprovedPremisesApplicationEntity).status = ApprovedPremisesApplicationStatus.REJECTED
     }
   }
+
+  fun applicationGetStatusFromArrivalDate(
```

Comment:
As we're only using the arrival date here, we could just pass the arrival date in